File: Beecrowd/1914 De Quem é a Vez/main.py
n = int(input())

for _ in range(n):

    a, b, c, d = [_ for _ in input().split()]
    n, m = [int(_) for _ in input().split()]
    
    jog = {}

    jog[b] = a
    jog[d] = c

    res = n + m

    if res % 2 == 0:
        escolha = "PAR"
    else:
        escolha = "IMPAR"

    # print(jog[escolha])

    for key in jog:
        if escolha == key:
            print(jog[key])

File: Beecrowd/1914 De Quem é a Vez/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        int qt = sc.nextInt();

        String[] jog1 = new String[2];
        String[] jog2 = new String[2];

        for (int i = 0; i < qt; i++) {
            jog1[0] = sc.next();
            jog1[1] = sc.next();
            jog2[0] = sc.next();
            jog2[1] = sc.next();

            int n = sc.nextInt();
            int m = sc.nextInt();

            int res = n + m;

            String escolha = "IMPAR";

            if (res % 2 == 0)
                escolha = "PAR";

            if (escolha.equals(jog1[1]))
                System.out.println(jog1[0]);
            else
                System.out.println(jog2[0]);
        }

        sc.close();
    }
}
